src/main/java/com/cloudbees/hudson/plugins/folder/icons/BuildStatusFolderIcon.java

```diff
@@ -0,0 +1,84 @@
+package com.cloudbees.hudson.plugins.folder.icons;
+
+import com.cloudbees.hudson.plugins.folder.Folder;
+import com.cloudbees.hudson.plugins.folder.FolderIcon;
+import com.cloudbees.hudson.plugins.folder.FolderIconDescriptor;
+import hudson.Extension;
+import hudson.model.*;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.Stapler;
+
+public class BuildStatusFolderIcon extends FolderIcon {
```

Comment:
I think it would really help clarify if you added a javadoc comment explaining how this will report status, e.g. what is the expected status when you have different child jobs.